File: test_art.js
function generateRandomArt(fingerprintBytes) {
    const width = 17;
    const height = 9;
    const grid = Array(width * height).fill(0);
    
    let x = 8;
    let y = 4;
    
    for (let i = 0; i < fingerprintBytes.length; i++) {
        let b = fingerprintBytes[i];
        for (let shift = 0; shift < 8; shift += 2) {
            const dir = (b >> shift) & 3;
            if ((dir & 1) === 0) x = Math.max(0, x - 1); else x = Math.min(width - 1, x + 1);
            if ((dir & 2) === 0) y = Math.max(0, y - 1); else y = Math.min(height - 1, y + 1);
            grid[y * width + x]++;
        }
    }
    
    const chars = " .o+=*BOX@%&#/^";
    let art = "+---[ECDSA 256]---+\n";
    for (let row = 0; row < height; row++) {
        art += "|";
        for (let col = 0; col < width; col++) {
            if (col === 8 && row === 4) art += "S";
            else if (col === x && row === y) art += "E";
            else {
                const val = grid[row * width + col];
                art += chars[Math.min(val, chars.length - 1)];
            }
        }
        art += "|\n";
    }
    art += "+----[SHA256]-----+";
    return art;
}
const crypto = require('crypto');
const hash = crypto.createHash('sha256').update('test_fingerprint_generation_for_ssh_random_art').digest();
console.log(generateRandomArt(hash));

File: test_art.py
import hashlib
def generateRandomArt(fingerprintBytes):
    width = 17
    height = 9
    grid = [0] * (width * height)
    x = 8
    y = 4
    for b in fingerprintBytes:
        for shift in range(0, 8, 2):
            dir = (b >> shift) & 3
            if (dir & 1) == 0: x = max(0, x - 1)
            else: x = min(width - 1, x + 1)
            if (dir & 2) == 0: y = max(0, y - 1)
            else: y = min(height - 1, y + 1)
            grid[y * width + x] += 1
            
    chars = " .o+=*BOX@%&#/^"
    art = "+---[ECDSA 256]---+\n"
    for row in range(height):
        art += "|"
        for col in range(width):
            if col == 8 and row == 4: art += "S"
            else:
                if col == x and row == y: art += "E"
                else:
                    val = grid[row * width + col]
                    art += chars[min(val, len(chars) - 1)]
        art += "|\n"
    art += "+----[SHA256]-----+"
    return art

h = hashlib.sha256(b"test_fingerprint_generation_for_ssh_random_art").digest()
print(generateRandomArt(h))
